packages/core/core/telemetry/record-span.ts

```diff
@@ -0,0 +1,27 @@
+import { Attributes, Span, Tracer } from '@opentelemetry/api';
+
+export function recordSpan<T>(
+  tracer: Tracer,
+  name: string,
+  attributes: Attributes,
+  fn: (span: Span) => Promise<T>,
+) {
+  return tracer.startActiveSpan(name, { attributes }, async span => {
+    try {
+      return await fn(span);
+    } catch (error) {
+      if (error instanceof Error) {
+        span.recordException({
+          name: error.name,
+          message: error.message,
+          stack: error.stack,
+        });
+      }
+      span.setStatus(2 as any); // SpanStatus.ERROR
```

Comment:
That's not going to work. I believe the actual API is something like `span.setStatus({code: 2, message: ...})`